docs/asdf/developer_overview.rst

```diff
@@ -0,0 +1,350 @@
+High level overview of the basic ASDF library
+=============================================
+
+This document is an attempt to make it easier to understand the design and
+workings of the python asdf library for those unfamiliar with it. This is
+expected to grow organically so at the moment it should not be considered
+complete or comprehensive.
+
+The design is somewhat complicated by the fact that in using pyyaml and
+jsonschema, one uses them by supplying code for these libraries to use, and
+without becoming familiar with how these packages work, the operation of the
+asdf internals isn't always very clear. This will try to provide a small amount
+of context for these package to illuminate how the code in asdf interacts with
+them.
+
+There are at least two ways of outlining the design. One is to give high level
+overviews of the various modules and how they interact with other modules. The
+other is to illustrate how code is actually invoked in common operations, this
+often being much more informative on a practical level (at least I find that to
+be the case). This document will attempt to do both.
+
+We will start with a high-level review of concepts and terms and point to where
+these are handled in the asdf modules. 
+
+Because of the complexity, this initial design overview will focus on issues of
+validation and tree construction when reading.
+
+Some terminology and definitions 
+--------------------------------
+
+**URI vs URL (Universal Resource Identifier)**. This is distinguished from URL
+(Universal Resource Locator) primarily in that URI is a mechanism for a unique
+name that follows a particular syntax, but itself may not indicate where the
+resource is. Generally URLs are expected to be used on the web for the HTTP
+protocol, though for asdf, this isn't necessarily the case as mentioned next.
+
+**Resolver:** Tools to map URIs and tags into actual locations of schema files,
+**which may be local directories (the usual approach) or an actual URL for
+**retrieval over the network. This is more complicated that it may seem for
+**reasons explained later.
```

Comment:
Remove the `**` from lines 38-40.

Reply:
These appeared when wrapping the text (editor was too clever). I thought I had caught all of these, but obviously not.